sql/core/src/test/scala/org/apache/spark/sql/execution/SparkSqlParserSuite.scala

```diff
@@ -61,6 +61,39 @@ class SparkSqlParserSuite extends AnalysisTest {
   private def intercept(sqlCommand: String, messages: String*): Unit =
     interceptParseException(parser.parsePlan)(sqlCommand, messages: _*)
 
+  test("Report Error for invalid usage of SET command") {
+    assertEqual("SET", SetCommand(None))
+    assertEqual("SET -v", SetCommand(Some("-v", None)))
+    assertEqual("SET spark.sql.key", SetCommand(Some("spark.sql.key" -> None)))
+    assertEqual("SET spark:sql:key=false", SetCommand(Some("spark:sql:key" -> Some("false"))))
+    assertEqual("SET spark.sql.    key=value",
```

Comment:
This is a bit weird. Is it possible to forbid it?

Reply:
hm, I'll try. But, it seems we already accept similar patterns, e.g.,
```
scala> sql("select * from default.t").show()
+---+                                                                           
|  a|
+---+
|  4|
|  5|
|  1|
|  2|
|  3|
+---+

scala> sql("select * from default   .t").show()
+---+
|  a|
+---+
|  4|
|  5|
|  1|
|  2|
|  3|
+---+

scala> sql("select * from default   .    t").show()
+---+
|  a|
+---+
|  4|
|  5|
|  1|
|  2|
|  3|
+---+
```
Should these also be prohibited? (But, I think it is hard to do so though...)

Reply:
I think table name is fine, but config name is really weird. It's kind of a silent behavior change as `SET spark.sql.    key=value` sets config `spark.sql.    key` before.

Reply:
https://www.antlr3.org/pipermail/antlr-interest/2007-October/024030.html

One way is to look at the hidden channel after parsing `configIdentifier`, and see if the next token is `WS`.